src/mlpack/methods/ann/activation_functions/silu_function.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file methods/ann/activation_functions/silu_function.hpp
+ * @author Fawwaz Mayda
+ *
+ * Definition and implementation of the Sigmoid Weighted Linear Unit function.
+ *
+ * For more information see the following paper
+ *
+ * @code
+ * @misc{elfwing2017sigmoidweighted ,
+ *    title = {Sigmoid-Weighted Linear Units for Neural Network Function Approximation in Reinforcement Learning},
+ *    author = {Stefan Elfwing and Eiji Uchibe and Kenji Doya},
+ *    year = {2017},
+ *    url = {https://arxiv.org/pdf/1702.03118.pdf},
+ *    eprint = {1702.03118},
+ *    archivePrefix = {arXiv},
+ *    primaryClass = {cs.LG} }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_SILU_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_SILU_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+
+/**
+ * The TanhExp function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& x * \frac{1}{1 + e^{-x}}\\
+ * f'(x) &=& \frac{1}{1 + e^{-x}} * (1 + x * (1-\frac{1}{1 + e^{-x}}))\\
+ * @f}
+ */
+    class SILUFunction {
```

Comment:
Looks like the indentation is off, you can take a look at other classes, that check what the indentation should look like.